pytorch_lightning/core/lightning.py

```diff
@@ -255,9 +256,13 @@ def _apply_batch_transfer_handler(
         self, batch: Any, device: Optional[torch.device] = None, dataloader_idx: int = 0
     ) -> Any:
         device = device or self.device
-        batch = self.on_before_batch_transfer(batch, dataloader_idx)
-        batch = self.transfer_batch_to_device(batch, device, dataloader_idx)
-        batch = self.on_after_batch_transfer(batch, dataloader_idx)
+        datahook_source = (
+            _DataHookSource(self, None) if self.trainer is None else self.trainer._data_connector._datahook_source
+        )
+
+        batch = datahook_source.get_hook("on_before_batch_transfer")(batch, dataloader_idx)
```

Comment:
Can you separate the two calls?

```suggestion
        hook = datahook_source.get_hook("on_before_batch_transfer")
        batch = hook(batch, dataloader_idx)
```